m2cgen/interpreters/interpreter.py

```diff
@@ -1,18 +1,64 @@
 import re
 
 from m2cgen import ast
+from m2cgen.interpreters.code_generator import BaseCodeGenerator
 
 
-class BaseInterpreter:
-
-    def __init__(self, cg, feature_array_name="input"):
-        self._cg = cg
-        self._feature_array_name = feature_array_name
+class BaseAstInterpreter:
```

Comment:
I really don't like `Ast` prefix here. It's kind of obvious what's being interpreted. Same true for `AstToCode...` why not just `CodeGenInterpreter`? Or even just `ToCodeInterpreter`?

Reply:
Done